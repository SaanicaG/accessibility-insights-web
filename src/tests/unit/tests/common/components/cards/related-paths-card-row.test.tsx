// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.
import { render } from '@testing-library/react';
import { RelatedPathsCardRow, RelatedPathsCardRowProps } from 'common/components/cards/related-paths-card-row';
import * as React from 'react';

describe(RelatedPathsCardRow.displayName, () => {
    it.each([[], null, undefined])(
        'renders as null with related paths: %p',
        (relatedPaths?: any) => {
            const props: RelatedPathsCardRowProps = {
                deps: null,
                index: 123,
                propertyData: relatedPaths,
            };
            const renderResult = render(<RelatedPathsCardRow {...props} />);

            expect(renderResult.container.firstChild).toBeNull();
        },
    );

    it('renders matching snapshot with related paths present', () => {
        const props: RelatedPathsCardRowProps = {
            deps: null,
            index: 123,
            propertyData: ['#path-1a;.path-1b', '#path-2', '.path-3'],
        };
        const renderResult = render(<RelatedPathsCardRow {...props} />);

        expect(renderResult.asFragment()).toMatchSnapshot();
    });
});
