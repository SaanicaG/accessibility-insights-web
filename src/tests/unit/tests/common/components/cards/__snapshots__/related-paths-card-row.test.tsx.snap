// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RelatedPathsCardRow renders matching snapshot with related paths present 1`] = `
<DocumentFragment>
  <mock-simplecardrow
    content="[object Object]"
    label="Related paths"
    rowkey="related-paths-row-123"
  />
</DocumentFragment>
`;
